Description: Avoid NULL pointer dereference in strncpy()
 Check for a pointer to a tag title and return if pointer is NULL.
Author: Marcus Meissner <marcus@jet.franken.de>
Origin: https://github.com/libexif/exif/commit/a702ad911f7c9824979a6534d87dfb1ec9928533
Bug: https://github.com/libexif/exif/issues/5
Last-Update: 2021-08-18

---
 exif/actions.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/exif/actions.c b/exif/actions.c
index 4fade01..d7ab870 100644
--- a/exif/actions.c
+++ b/exif/actions.c
@@ -715,7 +715,12 @@ show_entry_xml (ExifEntry *e, void *data)
 		fprintf (stdout, "%s", escape_xml(exif_entry_get_value (e, v, sizeof (v))));
 		fprintf (stdout, "</x%04x>", e->tag);
 	} else {
-		strncpy (t, exif_tag_get_title_in_ifd(e->tag, exif_entry_get_ifd(e)), sizeof (t));
+		const char *title = exif_tag_get_title_in_ifd(e->tag, exif_entry_get_ifd(e));
+		if (!title) {
+			/* might just be an unknown tag */
+			return;
+		}
+		strncpy (t, title, sizeof (t));
 		t[sizeof(t)-1] = 0;
 
 		/* Remove invalid characters from tag eg. (, ), space */
